src/librustdoc/html/render/mod.rs

```diff
@@ -2378,14 +2387,7 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
         }
 
         match ty {
-            clean::Type::ResolvedPath { did, .. } => {
-                let get_extern = || cache.external_paths.get(&did).map(|s| s.0.clone());
-                let fqp = cache.exact_paths.get(&did).cloned().or_else(get_extern);
-
-                if let Some(path) = fqp {
-                    out.push(path.join("::"));
-                }
-            }
+            clean::Type::ResolvedPath { did, .. } => process_path(did),
```

Comment:
Why change this to a closure?

Reply:
Because it's now used in two places: here, and for `Type::QPath` below. `Type::QPath` can't just use `work` to do it because it's now a `Path` not a `Type`.